WordPress/gutenberg
# PHPCS: Include the `test` directory

## What?
This PR adds the root-level test directory as a check target for PHPCS. At the same time, I added theme-specific rules and fix lint errors.

## Why?
Since the test directory already contains PHP files, I think it would make sense to include them.

## How?
The existing PHPCS rule causes the following error in the PHP files that the theme has:

- `Squiz.Commenting.FunctionComment.Missing`: Missing doc comment for function xxxxx()
- `Squiz.Commenting.FileComment.Missing`: Missing file doc comment
- `Squiz.Commenting.FileComment.WrongStyle`: You must use "/**" style comments for a file comment

These rules are also [excluded in the Theme Check plugin](https://github.com/WordPress/theme-check/blob/28c4137ef46469b56a4f8c843d12a83be0952920/.phpcs.xml.dist#L22-L35), so I excluded these rules as well. Then I fixed the remaining errors.

## Testing Instructions
In PHP files under the test directory, make changes that do not follow the rules. For example, remove the space after the opening parenthesis.

Run `npm run lint:php` and confirm that a Lint error occurs.
